crates/pathfinder/src/state.rs

```diff
@@ -0,0 +1,80 @@
+//! Contains the [GlobalStateTree] and [ContractsStateTree] trees, which combined
+//! store the total StarkNet state.
+//!
+//! These are abstractions built-on the [Binary Merkle-Patricia Tree](MerkleTree).
+
+use rusqlite::Transaction;
+
+use crate::{
+    core::{
+        ContractAddress, ContractRoot, ContractStateHash, GlobalRoot, StorageAddress, StorageValue,
+    },
+    state::merkle_tree::MerkleTree,
+};
+
+mod merkle_node;
+mod merkle_tree;
+
+/// A Binary Merkle-Patricia Tree which contains
+/// the storage state of all StarkNet contracts.
+struct ContractsStateTree<'a> {
+    tree: MerkleTree<'a>,
+}
+
+impl<'a> ContractsStateTree<'a> {
+    pub fn load(transaction: &'a Transaction, root: ContractRoot) -> anyhow::Result<Self> {
+        // TODO: move the string into storage.
```

Comment:
Since you added the TODO and we were last discussing more of a static approach to tables I'm thinking this could just be `&'static str` for the simplest alternative. But, later.